app/controllers/posts_controller.rb

```diff
@@ -94,6 +98,10 @@ def set_post
   end
   # Never trust parameters from the scary internet, only allow the white list through.
   def post_params
-    params.require(:post).permit(:title, :contents, :author_id)
+    @post_params ||= begin
+     p = params.require(:post).permit(:title, :contents, :author_id, :subjects)
+     p[:subjects] = Subjects.find(p[:subjects]) if p.include? :subjects
```

Comment:
Can we remove this and use `subject_ids` so that we don't have to fetch them from the db?